### PR TITLE
Refactor: rename env vars

### PR DESCRIPTION
**Pull request by kangmingtay**
## What kind of change does this PR introduce?
Refactor EXTERNAL_EMAIL_DISABLED & EXTERNAL_PHONE_DISABLED config vars to EXTERNAL_EMAIL_ENABLED and EXTERNAL_PHONE_ENABLED.

### CODE REVIEW

File: conf/configuration.go [hunk 1]
```diff
@@ -113,7 +113,7 @@ type MailerConfiguration struct {
 }
 
 type PhoneProviderConfiguration struct {
-	Disabled bool `json:"disabled"`
+	Enabled bool `json:"enabled"`
```
Comment (awalias) [inline conf/configuration.go:116]: should we explicitly set `default:"false"` here?

**Review by awalias — APPROVED** [review]
we need to make sure this one goes out on 2.0.0 or above with a warning about breaking change

**Comment by awalias** [comment]
and also make sure that we make the appropriate dashboard and db changes in the infrastructure repo

File: conf/configuration.go (hunk 1 above)
Comment (kangmingtay) [reply conf/configuration.go:116]: by default, it's automatically set to false. But yeah i should include that for readability.

**Comment by kangmingtay** [comment]
> and also make sure that we make the appropriate dashboard and db changes in the infrastructure repo

included in release notes [here](https://github.com/supabase/gotrue/releases/tag/v2.0.1)

**Pull request merged**